### PR TITLE
fixed unsubscribe bug for issue #938

### PR DESCRIPTION
**Pull request by jzlxiaohei**

### CODE REVIEW

File: src/createStore.js [hunk 1]
```diff
@@ -62,6 +62,7 @@ export default function createStore(reducer, initialState) {
 
     return function unsubscribe() {
       var index = listeners.indexOf(listener);
+      if (index === -1)return;
```
Comment (gaearon) [inline src/createStore.js]: Please be consistent with the project style.
In this case I'd rather write it as

```js
if (index !== -1) {
  listeners.splice(index, 1);
}
```

**Comment by gaearon** [comment]
Thank you for the PR! I left a nit, will be happy to merge after it's fixed.

File: src/createStore.js (hunk 1 above)
Comment (johanneslumpe) [inline src/createStore.js]: Nitpick: `index > -1` - saved 2 chars!

Comment (gaearon) [inline src/createStore.js]: I don't mind either way. :stuck_out_tongue: 
I was kind of a fan of `> -1` for some time, but now I think `!== -1` makes the intention clearer.
But whatever, I'm fine with either.

**Comment by gaearon** [comment]
I changed your fix to cover one more edge case: https://github.com/rackt/redux/commit/b7031ce3acb23b6ecadbd977b1cfa32486447904

**Comment by jzlxiaohei** [comment]
nice  :metal:

**Pull request merged**